spec/socialprovider.spec.js

```diff
@@ -102,6 +102,19 @@ describe("Tests for message batching in Social provider", function() {
     expect(xmppSocialProvider.client.send).toHaveBeenCalled();
   });
 
+  it("sends message to correct destinations", function() {
+    xmppSocialProvider.sendMessage('Bob', 'Hi', function() {});
+    xmppSocialProvider.sendMessage('Alice', 'Hi', function() {});
+    expect(xmppSocialProvider.client.send).not.toHaveBeenCalled();
+    jasmine.clock().tick(100);
+    expect(xmppSocialProvider.client.send.calls.count()).toEqual(2);
+    var dest = xmppSocialProvider.client.send.calls.first().args[0].parent.attrs.to;
+    if (dest !== 'Bob' && dest !== 'Alice') {
+      expect(dest).toEqual('Bob');
```

Comment:
I think it would show up at the end of the jasmine run as 'an error was
encountered', but not fail the given test case, due to issues with
accounting for asynchronicity.

On Thu Jan 08 2015 at 8:46:02 AM dborkan <notifications@github.com> wrote:

> In spec/socialprovider.spec.js
> <https://github.com/freedomjs/freedom-social-xmpp/pull/80#discussion-diff-22663280>
> :
>
> > @@ -102,6 +102,19 @@ describe("Tests for message batching in Social provider", function() {
> >      expect(xmppSocialProvider.client.send).toHaveBeenCalled();
> >    });
> >
> > +  it("sends message to correct destinations", function() {
> > +    xmppSocialProvider.sendMessage('Bob', 'Hi', function() {});
> > +    xmppSocialProvider.sendMessage('Alice', 'Hi', function() {});
> > +    expect(xmppSocialProvider.client.send).not.toHaveBeenCalled();
> > +    jasmine.clock().tick(100);
> > +    expect(xmppSocialProvider.client.send.calls.count()).toEqual(2);
> > +    var dest = xmppSocialProvider.client.send.calls.first().args[0].parent.attrs.to;
> > +    if (dest !== 'Bob' && dest !== 'Alice') {
> > +      expect(dest).toEqual('Bob');
>
> Not sure - if you throw an exception with a string "Expected dest to be
> Bob or Alice but ", would that show up in the console as that 1 test case
> failing? Or would it break all later test cases as well?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/freedomjs/freedom-social-xmpp/pull/80/files#r22663280>
> .
>